### DISCUSSION TITLE
PHP Digest Authentication - page is not passing username and password

### DISCUSSION THREAD

**Discussion by JasonTonGit** [AxisCommunications/discussions#161] [General] [answered]
I am trying to write a simple php page that auto loads a camera stream in an iframe. The page is working except it is still asking me for a username and password and I'm not sure why these aren't being passed to the camera. I have tried this multiple ways and get the same result. Here is one of the code examples:

```

<?php
    // Set up login credentials
    $username = "root";
    $password = "password";

    // Set up camera IP and resolution
    $camera_ip = "192.168.1.5";
    $resolution = "320x240";

    // Build the video feed URL
    $video_feed_url = "http://$camera_ip/axis-cgi/mjpg/video.cgi?resolution=$resolution";

    // Set up digest authentication
    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL, $video_feed_url);
    curl_setopt($ch, CURLOPT_HTTPAUTH, CURLAUTH_DIGEST);
    curl_setopt($ch, CURLOPT_USERPWD, "$username:$password");
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
    $output = curl_exec($ch);
    curl_close($ch);

    // Create the iframe
    echo "<iframe src='$video_feed_url' width='320' height='240' frameborder='0'></iframe>";
?>

```

**Comment by onie5** [comment]
Just a suggestion, although I tried this in C not php.

Set your curl options for auth as CURLAUTH_ANY.

For credentials, try to separate the username/password by using CURLOPT_USERNAME and CURLOPT_PASSWORD

**Reply by JasonTonGit** [reply]
Ya, I had the same thoughts and tried the below, but I'm getting the same thing.  Any other thoughts?  Any help would be greatly appreciated....

```
<?php
echo test;
    // Set up login credentials
    $username = "root";
    $password = "chEve7Pu";

    // Set up camera IP and resolution
    $camera_ip = "10.126.177.45";
    $resolution = "320x240";

    // Build the video feed URL
    $video_feed_url = "http://$camera_ip/axis-cgi/mjpg/video.cgi?resolution=$resolution";

    // Set up digest authentication
    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL, $video_feed_url);
    curl_setopt($ch, CURLOPT_HTTPAUTH, CURLAUTH_ANY);
	curl_setopt($ch, CURLOPT_USERNAME, "$username");
	curl_setopt($ch, CURLOPT_PASSWORD, "$password");
    //curl_setopt($ch, CURLOPT_USERPWD, "$username:$password");
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
    $output = curl_exec($ch);
    curl_close($ch);

    // Create the iframe
    echo "<iframe src='$video_feed_url' width='320' height='240' frameborder='0'></iframe>";
?>
```

**Comment by janssen70** [comment, marked as answer]
The php code runs server-side, meaning on server-side you can do authenticated requests
The iframe src-attribute is processed by the browser, it will access the camera directly but it does not have the credentials for the camera
For this application to work you'll need to enable guest viewing on the camera

In older days you could add the credentials in the url but that's not possible anymore, for cybersecurity reasons
